search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalApiResponseBuilder.kt

```diff
@@ -0,0 +1,241 @@
+            total,
+            resourceUrl,
+            query.entitiesQuery.paginationQuery,
+            requestParams,
+            mediaType,
+            contexts
+        )
+        val temporalQuery = query.temporalQuery
+        if (temporalQuery.isLastNInsideLimit()) {
+            return successResponse
+        }
+
+        val attributesWhoReachedLimit = getAttributesWhoReachedLimit(entities, query)
+
+        if (attributesWhoReachedLimit.isEmpty()) {
+            return successResponse
+        }
+
+        val range = getTemporalPaginationRange(attributesWhoReachedLimit, query)
+
+        val filteredEntities = entities.map { filterEntityInRange(it, range, query) }
+        return ResponseEntity.status(HttpStatus.PARTIAL_CONTENT).apply {
+            this.headers(
+                successResponse.headers
+            )
+            this.header(
+                HttpHeaders.CONTENT_RANGE,
+                getHeaderRange(range, query.temporalQuery)
+            )
+        }.body(serializeObject(filteredEntities.toFinalRepresentation(ngsiLdDataRepresentation)))
+    }
+
+    fun buildEntityTemporalResponse(
+        entity: CompactedEntity,
+        query: TemporalEntitiesQuery,
+        mediaType: MediaType,
+        requestParams: MultiValueMap<String, String>,
+        contexts: List<String>,
+    ): ResponseEntity<String> {
+        val ngsiLdDataRepresentation = parseRepresentations(requestParams, mediaType)
+        val successResponseHeader = prepareGetSuccessResponseHeaders(mediaType, contexts)
+
+        if (query.temporalQuery.isLastNInsideLimit()) {
+            return successResponseHeader.body(
+                serializeObject(
+                    entity.toFinalRepresentation(ngsiLdDataRepresentation)
+                )
+            )
+        }
+
+        val attributesWhoReachedLimit = getAttributesWhoReachedLimit(listOf(entity), query)
+
+        if (attributesWhoReachedLimit.isEmpty()) {
+            return successResponseHeader.body(
+                serializeObject(
+                    entity.toFinalRepresentation(ngsiLdDataRepresentation)
+                )
+            )
+        }
+
+        val range = getTemporalPaginationRange(attributesWhoReachedLimit, query)
+        val filteredEntity = filterEntityInRange(entity, range, query)
+
+        return ResponseEntity.status(HttpStatus.PARTIAL_CONTENT)
+            .apply {
+                this.header(
+                    HttpHeaders.CONTENT_RANGE,
+                    getHeaderRange(range, query.temporalQuery)
+                )
+                this.headers(
+                    successResponseHeader.body("").headers
+                )
+            }.body(
+                serializeObject(
+                    filteredEntity.toFinalRepresentation(ngsiLdDataRepresentation)
+                )
+            )
+    }
+
+    private fun getAttributesWhoReachedLimit(entities: List<CompactedEntity>, query: TemporalEntitiesQuery): List<Any> {
```

Comment:
you also have to take into account:
- queries with aggregated values (see https://github.com/stellio-hub/ngsi-ld-test-suite/blob/feature/stellio-tcs-stack/data/temporalEntities/expectations/vehicle-temporal-representation-020-11-03-expectation.json for instance)
- entities whose attributes have multiple datasets:

```
{
    "id": "urn:ngsi-ld:Distribution:Zone:104-P-001:WithTemporalValues",
    "type": "Distribution",
    "attrWithDatasetId": [
        {
            "type": "Property",
            "datasetId": "urn:ngsi-ld:Dataset:K63:5To10Centimeters",
            "values": [
                [
                    15.0,
                    "2009-02-01T12:55:00Z"
                ],
                [
                    94.0,
                    "2009-02-02T08:55:00Z"
                ]
            ]
        },
        {
            "type": "Property",
            "datasetId": "urn:ngsi-ld:Dataset:K63:0To1Meter",
            "values": [
                [
                    15.0,
                    "2007-10-03T10:30:00Z"
                ],
                [
                    61.0,
                    "2007-10-06T06:56:00Z"
                ]
            ]
        }
    ]
}
```

and 

```
{
    "id": "urn:ngsi-ld:Distribution:Zone:104-P-001:WithNormalizedRepresentation",
    "type": "Distribution",
    "attrWithDatasetId": [
        {
            "type": "Property",
            "value": 15.0,
            "unitCode": "K63",
            "datasetId": "urn:ngsi-ld:Dataset:K63:5To10Centimeters",
            "instanceId": "urn:ngsi-ld:Instance:1b2db622-7c80-4dc9-beb6-7469c51bb1dd",
            "observedAt": "2009-02-01T12:55:00Z"
        },
        {
            "type": "Property",
            "value": 94.0,
            "unitCode": "K63",
            "datasetId": "urn:ngsi-ld:Dataset:K63:5To10Centimeters",
            "instanceId": "urn:ngsi-ld:Instance:f3b6fc21-bb44-49ad-88d8-06a4aab4c38e",
            "observedAt": "2009-02-02T08:55:00Z"
        },
        {
            "type": "Property",
            "value": 15.0,
            "unitCode": "K63",
            "datasetId": "urn:ngsi-ld:Dataset:K63:0To1Meter",
            "instanceId": "urn:ngsi-ld:Instance:3665309a-f921-4803-a9c9-a4066f4e4092",
            "observedAt": "2009-01-30T11:28:00Z"
        },
        {
            "type": "Property",
            "value": 380.0,
            "unitCode": "K63",
            "datasetId": "urn:ngsi-ld:Dataset:K63:0To1Meter",
            "instanceId": "urn:ngsi-ld:Instance:0c3d08ad-4359-4e13-9b55-e75bac2cc98b",
            "observedAt": "2009-02-02T08:55:00Z"
        }
    ]
}
```